### ISSUE TITLE
Fix extract_loss regex to capture scientific-notation losses

### ISSUE THREAD

**Issue by msalganik** [niznik-dev/cruijff_kit#448]
## For Mattie

**Problem.** `extract_loss.final_loss()` parses training loss out of torchtune SLURM logs to populate the "Final Loss" column in `summary.md`. Its regex captures the digits-and-dots part of a number but silently drops scientific-notation exponents. So a real loss of `6.05e-06` (effectively zero — model fully saturated) gets reported as `6.05` (looks broken).

**Why fix it.** Caught by hand on the consistency-curve pilot just now: the N=1000 model achieves 99.1% accuracy, training loss in the slurm log is `~6e-6`, but `summarize-experiment` initially wrote `6.05` into the summary table. I had to manually inspect the slurm log and override the value. Anyone reading only `summary.md` from a saturated run would conclude "loss is high, training failed" — the exact opposite of the truth. The full consistency-curve experiment will hit this on every fine-tuned run that learns the rule.

**What changes after the fix.**
- `summary.md` shows the actual final loss for every run regardless of magnitude.
- New regression tests cover scientific-notation losses (currently uncovered — the existing fixture only has decimal-format losses, which is why the bug shipped).
- No behavioral change for runs that emit decimal-format losses (the fixture run + every other already-tested case continues to parse identically).

**Acceptance.** With these changes:
1. `extract_losses` and `final_loss` correctly parse `Loss: 6.0498432503663935e-06` to `~6e-6` (not `6.05`).
2. Three new test cases cover `e-06`, `e+02`, and a log that mixes both decimal and sci-notation losses.
3. All existing tests in `test_summarize_fixtures.py` still pass unchanged.

**Effort estimate.** ~1 line of source + ~25 lines of tests in one file. Tiny.

---

### Investigation already done

Don't re-investigate — go straight to the patch. Key findings:

- `src/tools/torchtune/extract_loss.py:9` — the regex is `(\d+)\|(\d+)\|Loss: ([0-9.]+)`. The third group is digits-and-dots only — no `e`, `+`, `-`.
- The custom torchtune recipe formats losses with Python's default `repr()`, which switches to scientific notation when |value| < 1e-4. Once a model converges, **every** subsequent loss line uses sci-notation.
- Concrete observation from a real run: the last loss line in `/scratch/gpfs/MSALGANIK/mjs3/ck-experiments/consistency_curve_pilot_nth14_k20/Llama-3.2-3B-Instruct_N1000/artifacts/slurm-7604383.out` is `Loss: 6.0498432503663935e-06`. `final_loss()` currently returns `(3, 750, 6.0498432503663935)` — losing 6 orders of magnitude.
- Only callers of these functions: `tests/unit/test_summarize_fixtures.py` (lines 12-15) and `.claude/skills/summarize-experiment/SKILL.md` (the skill's documented invocation). No production call sites beyond `summarize-experiment`.
- The fixture `tests/fixtures/summarize/slurm_training_output.txt` contains zero scientific-notation losses — confirmed via `grep -E "Loss: [0-9.]+e[+-][0-9]+"`. That's why the bug shipped.
- `float()` already accepts scientific notation natively, so no parsing change is needed in `extract_losses` — only the regex needs widening.

### Files to change

Source (1 file):
1. `src/tools/torchtune/extract_loss.py` — widen the third capture group.

Tests (1 modified, 0 new):
2. `tests/unit/test_summarize_fixtures.py` — append three tests inside the existing `TestLossExtraction` class (currently ends ~line 75).

Adding a sci-notation block to the existing fixture file would also work, but inline-string test inputs are cheaper, easier to read, and keep the change to one file.

### Patch 1: `src/tools/torchtune/extract_loss.py`

Replace line 9:

```python
# Before
LOSS_PATTERN = re.compile(r"(\d+)\|(\d+)\|Loss: ([0-9.]+)")
# After
LOSS_PATTERN = re.compile(r"(\d+)\|(\d+)\|Loss: ([0-9.eE+-]+)")
```

The `[0-9.eE+-]+` character class is intentionally permissive — torchtune emits well-formed numbers, and any malformed match would raise `ValueError` from `float()` in `extract_losses` (a healthy fail-loud, since malformed input means something is wrong upstream).

### Patch 2: `tests/unit/test_summarize_fixtures.py`

Append three methods to the existing `TestLossExtraction` class:

```python
    def test_extracts_scientific_notation_loss(self):
        """Converged-model logs emit losses in sci-notation; the regex must capture them."""
        line = "3|750|Loss: 6.0498432503663935e-06: 100%|##########| 250/250 [12:05<00:00,  2.90s/it]"
        losses = extract_losses(line)
        assert len(losses) == 1
        assert losses[0][2] == pytest.approx(6.0498e-6, rel=1e-4)

    def test_extracts_positive_exponent_loss(self):
        """Edge case: very large losses (early training, divergence) may use e+ notation."""
        line = "1|1|Loss: 2.5e+02: starting"
        losses = extract_losses(line)
        assert len(losses) == 1
        assert losses[0][2] == pytest.approx(250.0)

    def test_mixed_decimal_and_sci_notation(self):
        """A realistic converging-then-converged log mixes both formats."""
        text = (
            "1|10|Loss: 1.234: ...\n"
            "2|20|Loss: 5.6e-05: ...\n"
            "3|30|Loss: 7.89e-08: ...\n"
        )
        losses = extract_losses(text)
        assert [step for _, step, _ in losses] == [10, 20, 30]
        assert losses[0][2] == pytest.approx(1.234)
        assert losses[1][2] == pytest.approx(5.6e-5)
        assert losses[2][2] == pytest.approx(7.89e-8)
        assert final_loss(text)[2] == pytest.approx(7.89e-8)
```

### Test plan

Run from the repo root with the `cruijff` conda env active.

**a. Unit tests (all pass; new tests cover the three sci-notation cases):**
```bash
pytest tests/unit/test_summarize_fixtures.py -v
```
Expected: all tests pass. The existing tests rely on the fixture file (decimal-only losses), so the regex widening leaves them unchanged.

**b. Replay against the real-world failing log (before/after):**
```bash
python -c "
from cruijff_kit.tools.torchtune.extract_loss import final_loss
path = '/scratch/gpfs/MSALGANIK/mjs3/ck-experiments/consistency_curve_pilot_nth14_k20/Llama-3.2-3B-Instruct_N1000/artifacts/slurm-7604383.out'
print(final_loss(open(path).read()))
"
# Before fix:  (3, 750, 6.0498432503663935)
# After fix:   (3, 750, 6.0498432503663935e-06)
```
Confirms the patch closes the actual incident, not just an abstract case.

**c. Sanity check that the existing fixture still parses identically:**
```bash
python -c "
from pathlib import Path
from cruijff_kit.tools.torchtune.extract_loss import final_loss, extract_losses
text = Path('tests/fixtures/summarize/slurm_training_output.txt').read_text()
print('count:', len(extract_losses(text)), '(expected 18)')
print('final:', final_loss(text), '(expected (epoch, 500, ~0.26427))')
"
```
Expected: count=18, final loss approximately 0.26427456736564636. (These are the values asserted in `test_total_loss_count` and `test_final_loss_returns_last_match`.)

**d. Lint:**
```bash
ruff check src/tools/torchtune/extract_loss.py tests/unit/test_summarize_fixtures.py
ruff format --check src/tools/torchtune/extract_loss.py tests/unit/test_summarize_fixtures.py
```

### PR conventions

- Branch: `<issue-number>-extract-loss-sci-notation`.
- PR body starts with `Closes #<issue-number>` (per `.github/pull_request_template.md`).
- Include in the Testing Instructions section: pytest output for (a) and the before/after one-liner output from (b).

### Out of scope (do NOT do in this PR)

- Reformatting torchtune's loss output to suppress sci-notation (would need a recipe patch and hurts log readability).
- Retroactively re-running `summarize-experiment` on prior experiments to fix their `summary.md` files. Each user can re-run summarize-experiment on demand; not this PR's job.
- Handling `NaN` / `Inf` losses — separate issue if/when it comes up.

---

— Claude x Matt

**Comment by msalganik** [comment]
@niznik-dev I've put this into the next milestone because it is needed for experiments with model organisms.  Those have loss close to 0, which gets expessed in scientific notation, which confuses claude give the current workflow.  This is a small fix to a problem that I've not seen until we had model organisms.
